tests/lib/nested.sh

```diff
@@ -154,7 +154,7 @@ get_nested_snap_rev(){
 get_snap_rev_for_channel(){
     SNAP=$1
     CHANNEL=$2
-    execute_remote "snap info $SNAP" | grep "$CHANNEL" | awk '{ print $4 }' | sed 's/.*(\(.*\))/\1/' | tr -d '\n'
+    snap info "$SNAP" | grep "$CHANNEL" | awk '{ print $4 }' | sed 's/.*(\(.*\))/\1/' | tr -d '\n'
```

Comment:
I don't understand this change, shouldn't we be getting the revision info the system under test? I don't know why it might be different but it could be, couldn't it ?

Reply:
Well, yes but I had to do this because I need to get that info before the system is started to see it we can skip the test. 
As nested just works on amd64 this should work well now. what I can do it so detect the nested architecture and based on that get the revision. Does it make sense?

Reply:
I suppose that since this function is actually only used in this one test it's fine, but it's a bit confusing for the caller, considering most of the other helpers in nested.sh are things that are executed remotely. Maybe this helper could be moved to somewhere like snaps.sh in a followup since there's nothing that's specifically nested or remote about this helper function anymore?